camel-dhis2-api/src/test/java/org/apache/camel/component/dhis2/api/Dhis2GetTestCase.java

```diff
@@ -156,10 +164,21 @@ public void close() {
 
     @Test
     public void testCollectionGivenPagingIsTrue() {
-        Dhis2Response dhis2Response = new Dhis2Response() {
+
+        Request request = new Request.Builder().url("http://localhost").build();
+        Response response = new Response.Builder()
+            .request(request)
+            .protocol(Protocol.HTTP_1_1)
+            .code(200)
+            .message("OK")
+            .body(ResponseBody.create("", MediaType.get("application/json")))
+            .build();
+
+        ConverterFactory converterFactory = mock(ConverterFactory.class);
+        DefaultDhis2Response defaultDhis2Response = new DefaultDhis2Response(response, converterFactory) {
```

Comment:
This leads to a cast error since the `Dhis2Response` is cast to a `DefaultDhis2Response` in the PageIterable class  in the sdk to be able to fetch the url:
```
public PageIterable( String collectionName, ConverterFactory converterFactory,
        OkHttpClient httpClient, Class<T> responseType, Dhis2Response dhis2Response )
    {
        Page firstPage = dhis2Response.returnAs( Page.class );
        List<T> collection = (List<T>) converterFactory.createResponseConverter( responseType )
            .convert( (List<Map<String, Object>>) firstPage.getAdditionalProperties().get( collectionName ) );
        this.dhis2Response = dhis2Response;
        this.url = ((DefaultDhis2Response) dhis2Response).getResponse().request().url().toString();
        this.currentIterator = collection.iterator();

        this.currentPage = firstPage;
        this.collectionName = collectionName;
        this.converterFactory = converterFactory;
        this.httpClient = httpClient;
        this.responseType = responseType;
        this.lastPage = false;
    }
```
Might have to do some refactoring in the PageIterable class to prevent this.